core/src/test/java/org/infinispan/distribution/DistSyncFuncTest.java

```diff
@@ -306,27 +306,19 @@ public void testComputeFromNonOwner() throws InterruptedException {
       // compute function applied
       initAndTest();
       Object retval = getFirstNonOwner("k1").compute("k1", (k, v) -> "computed_" + k + "_" + v);
-      if (testRetVals) assertEquals("computed_k1_value", retval);
       asyncWait("k1", ComputeCommand.class);
-      assertOnAllCachesAndOwnership("k1", "computed_k1_value");
+      if (testRetVals) assertEquals("computed_k1_value", retval);
+      eventuallyAssertOnAllCachesAndOwneship("k1", "computed_k1_value");
```

Comment:
The flakiness might demonstrate a problem in the code that's not handled. Tests should not be fixed by loosening the requirements unless we know what's going on.